src/main/java/org/mastodon/mamut/feature/branch/BranchSpotDoubleFeatureComputer.java

```diff
@@ -0,0 +1,73 @@
+package org.mastodon.mamut.feature.branch;
+
+import net.imglib2.util.LinAlgHelpers;
+import org.mastodon.mamut.feature.CancelableImpl;
+import org.mastodon.mamut.feature.MamutFeatureComputer;
+import org.mastodon.mamut.model.Model;
+import org.mastodon.mamut.model.Spot;
+import org.mastodon.mamut.model.branch.BranchSpot;
+import org.scijava.plugin.Parameter;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+public abstract class BranchSpotDoubleFeatureComputer extends CancelableImpl implements MamutFeatureComputer
+{
+	@Parameter
+	protected Model model;
+
+	@Override
+	public void run()
+	{
+		super.deleteCancelReason();
+		computeAll( getOutput() );
+	}
+
+	protected abstract DoublePropertyFeature< BranchSpot > getOutput();
+
+	private void computeAll( final DoublePropertyFeature< BranchSpot > output )
+	{
+		for ( BranchSpot branchSpot : model.getBranchGraph().vertices() )
+		{
+			if ( isCanceled() )
+				break;
+			output.map.set( branchSpot, computeValue( branchSpot ) );
+		}
+	}
+
+	protected abstract double computeValue( final BranchSpot branchSpot );
+
+	protected Iterator< Spot > getSpotIterator( final BranchSpot branchSpot )
+	{
+		return model.getBranchGraph().vertexBranchIterator( branchSpot );
```

Comment:
Remember to call model.getBranchGraph().releaseIterator(...) after you are don't using the iterator.

Reply:
Thanks for spotting this. It is so easy to make this wrong. And I did it wrong (again). I fixed the problem and added a sentence in the javadoc to make readers aware of this.